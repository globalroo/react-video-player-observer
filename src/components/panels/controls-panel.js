import React from "react";
import PropTypes from "prop-types";

export const createClickHandler = ({ loadAsset, asset }) => () => loadAsset(asset);

export const MovieButton = ({ asset, loadAsset }) => {
	const clickHandler = createClickHandler({ loadAsset, asset });
	return (
		<div className="btn-group" role="group">
			<button type="button" onClick={clickHandler} className="btn btn-info">{asset.name}</button>
		</div>
	);
};

MovieButton.propTypes = {
	loadAsset: PropTypes.func,
	asset: PropTypes.object
};

export const ControlsPanel = ({ availableMedia, loadAsset, asset }) => (
	<div className="panel panel-default">
		<div className="panel-heading">
			<h4>Controls</h4>
		</div>
		<div className="panel-body">
			<div className="btn-group btn-group-vertical btn-group-max-width" role="group" aria-label="Switch movies">
				{
					availableMedia.map((asset, ix) =>
						<MovieButton key={`asset_${ix}`} asset={asset} loadAsset={loadAsset} />)
				}
			</div>
		</div>
		<div className="panel-footer">
			{(asset && asset.name) && <small>Now playing: <span className="now-playing">{asset.name}</span></small>}
		</div>
	</div>
);

ControlsPanel.propTypes = {
	availableMedia: PropTypes.array,
	loadAsset: PropTypes.func,
	asset: PropTypes.object
};
