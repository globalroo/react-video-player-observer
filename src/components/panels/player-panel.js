import { Player } from "src/components/player/player";
import React from "react";
import PropTypes from "prop-types";

export const PlayerPanel = ({ onPlayerLoaded }) => (
	<div className="panel panel-default">
		<div className="panel-heading">
			<h4>Video Player</h4>
		</div>
		<div className="panel-body">
			<Player onPlayerLoaded={onPlayerLoaded} />
		</div>
	</div>
);

PlayerPanel.propTypes = {
	onPlayerLoaded: PropTypes.func,
};
