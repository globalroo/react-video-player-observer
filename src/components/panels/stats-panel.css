.latestEvent {
	font-size:130%;
}
.halfColumn {
	width: 50%;
}
