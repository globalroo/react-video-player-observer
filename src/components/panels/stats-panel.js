import "./stats-panel.css";
import React from "react";
import PropTypes from "prop-types";

export const StatsPanel = ({ playerState, eventHistory }) => {
	const currentTime = (playerState && playerState.currentTime) ? playerState.currentTime.toFixed(3) : 0;
	const remainingTime = (playerState && playerState.remainingTime) ? playerState.remainingTime.toFixed(3) : 0;
	const stateClass = ( playerState.error.code !== 0) ? "danger" : "info";
	return (
		<div className="panel panel-default">
			<div className="panel-heading">
				<h4>Observer</h4>
			</div>
			<div className="panel-body">
				<table className="table table-striped table-responsive">
					<tbody>
						<tr>
							<td className="halfColumn">currentTime</td>
							<td className="halfColumn">{currentTime}</td>
						</tr>
						<tr>
							<td>remainingTime</td>
							<td>{remainingTime}</td>
						</tr>
						<tr>
							<td>error?</td>
							<td className={stateClass}>{playerState.error.description} [{playerState.error.code}]</td>
						</tr>
						{
							eventHistory.map(({ event, timeStamp }, ix) => {
								const className = (ix === 0) ? "success latestEvent" : "";
								return (
									<tr className={className} key={ix} >
										<td>{event}</td>
										<td>{timeStamp}</td>
									</tr>
								);
							})
						}
					</tbody>
				</table>
			</div>
		</div>
	);
};

StatsPanel.propTypes = {
	playerState: PropTypes.object,
	eventHistory: PropTypes.array
};
