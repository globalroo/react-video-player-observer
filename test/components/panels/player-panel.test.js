import React from "react";
import * as renderer from "react-test-renderer";
import { PlayerPanel } from "src/components/panels/player-panel";

jest.mock("src/components/player/player", () => ({
	// eslint-disable-next-line react/prop-types
	Player: ({ onPlayerLoaded }) => (
		<div> Player component created and callback passed through to Player with value:
			{onPlayerLoaded()}
		</div>
	)
}));

describe("Player component tests", () => {
	test("Player component renders as expected", () => {
		const component = renderer.create(
			<PlayerPanel onPlayerLoaded={ () => ("mockCallback") } />
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
});
