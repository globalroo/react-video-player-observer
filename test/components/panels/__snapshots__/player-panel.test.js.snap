// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player component tests Player component renders as expected 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    <h4>
      Video Player
    </h4>
  </div>
  <div
    className="panel-body"
  >
    <div>
       Player component created and callback passed through to Player with value:
      mockCallback
    </div>
  </div>
</div>
`;
